packages/data/src/types.ts

```diff
@@ -6,7 +6,7 @@ import type { combineReducers as reduxCombineReducers } from 'redux';
 
 type MapOf< T > = { [ name: string ]: T };
 
-export type ActionCreator = Function | Generator;
+export type ActionCreator = ( ...args: any[] ) => any | Generator;
```

Comment:
`Function` doesn't let you add return types and isn't compatible with the arrow syntax, so it needs to be removed so that we can wrap around the type below.